### PR TITLE
Fix entrypoint to respect CHE_DOCKER_IP

### PR DESCRIPTION
**Pull request by ghost**
### What does this PR do?

Fix for Che entrypoint so that CHE_DOCKER_IP is respected if provided

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/5449

### CODE REVIEW

**Review by TylerJewell — APPROVED** [review]

File: dockerfiles/che/entrypoint.sh [hunk 1]
```diff
@@ -83,8 +83,10 @@ set_environment_variables () {
   ### Set value of derived environment variables.
 
   # CHE_DOCKER_IP is used internally by Che to set its IP address
-  if [[ -n "${CHE_IP}" ]]; then
-    export CHE_DOCKER_IP="${CHE_IP}"
+  if [ -z "${CHE_DOCKER_IP}" ];then
```
Comment (benoitf) [inline dockerfiles/che/entrypoint.sh]: (edited I miss read)
 in bash, we've to use `[[ ]]` and there is a missing space after ;

**Comment by benoitf** [comment]
@eivantsov sorry you were right on z flag

File: dockerfiles/che/entrypoint.sh (hunk 1 above)
Comment (garagatyi) [reply dockerfiles/che/entrypoint.sh]: Amazing answer https://stackoverflow.com/questions/3869072/test-for-non-zero-length-string-in-bash-n-var-or-var/3870055#3870055

**Review by garagatyi — APPROVED** [review]

File: dockerfiles/che/entrypoint.sh [hunk 2]
```diff
@@ -83,8 +83,10 @@ set_environment_variables () {
   ### Set value of derived environment variables.
 
   # CHE_DOCKER_IP is used internally by Che to set its IP address
-  if [[ -n "${CHE_IP}" ]]; then
-    export CHE_DOCKER_IP="${CHE_IP}"
+  if [[ -z "${CHE_DOCKER_IP}" ]];then
```
Comment (benoitf) [inline dockerfiles/che/entrypoint.sh]: missing space after ; and fine for me

**Review by benoitf — APPROVED** [review]

**Comment by slemeur** [comment]
@eivantsov : please assign milestone and labels when merging.

**Pull request merged**